src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -891,7 +892,14 @@ private IEnumerable<PortableExecutableReference> GetPortableExecutableReferences
 
         private void WriteTypes(Assembly assembly)
         {
-            WriteObject(assembly.GetTypes(), true);
+            foreach (Type type in assembly.GetTypes())
+            {
+                // We only write out types that are not auto-generated by compiler.
+                if (type.GetCustomAttribute<CompilerGeneratedAttribute>() is null)
```

Comment:
Can we use:
```suggestion
                if (type.IsDefined(typeof(CompilerGeneratedAttribute), inherit: false))
```

Reply:
I don't think there is much difference here. I will keep it as is.

Reply:
GetCustomAttribute() does extra allocations at least.